extensions/guacamole-vault/modules/guacamole-vault-ksm/src/main/java/org/apache/guacamole/vault/ksm/conf/KsmConfigurationService.java

```diff
@@ -176,6 +188,20 @@ public boolean getMatchUserRecordsByDomain() throws GuacamoleException {
         return environment.getProperty(MATCH_USER_DOMAINS, false);
     }
 
+    /**
+     * Return the minimum number of milliseconds between KSM API calls. If not
+     * otherwise configured, this value will be 60 seconds.
+     *
+     * @return
+     *     The minimum number of milliseconds between KSM API calls.
+     *
+     * @throws GuacamoleException
+     *     If the value specified within guacamole.properties cannot be
+     *     parsed or does not exist.
+     */
+    public long getKsmApiTimeout() throws GuacamoleException {
+        return environment.getProperty(KSM_API_CALL_TIMEOUT, 60000L);
```

Comment:
The original default timeout was 5 seconds, which means that under heavy load, an entire monthly API call budget could be used up in a bit over 3 hours (the base subscription includes 2500 calls / month).

At 60 seconds, the budget would stretch to nearly 2 days.

Reply:
I'm not sure about this. Defaulting to 60 seconds would mean people will not see changes made to records until a full minute has elapsed, which could be surprising behavior for an admin troubleshooting a connection.

Reply:
What interval would you suggest?

Reply:
Eh, I'll go to 10. That will at least buy a little bit more time before a monthly allowance will run out without being too slow to refresh.